### PR TITLE
Remove remaining isSwiftVersion3() checks

### PR DESCRIPTION
**Pull request by slavapestov**
Progress on rdar://43101816.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

File: lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -364,8 +364,7 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
 
   Opts.EnableSwift3ObjCInference =
     Args.hasFlag(OPT_enable_swift3_objc_inference,
-                 OPT_disable_swift3_objc_inference,
-                 Opts.isSwiftVersion3());
+                 OPT_disable_swift3_objc_inference, false);
```
Comment (DougGregor) [inline lib/Frontend/CompilerInvocation.cpp]: I hope we can kill this, too. Is that a separate task?

**Review by DougGregor — APPROVED** [review]

File: lib/Frontend/CompilerInvocation.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/Frontend/CompilerInvocation.cpp]: Yeah, I’d like to consider it separate since it’s pretty self contained.

**Pull request merged**